src/Microsoft.ML.SearchSpace/Option/ChoiceOption.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Linq;
+
+#nullable enable
+
+namespace Microsoft.ML.SearchSpace.Option
+{
+    public class ChoiceOption : OptionBase
```

Comment:
Are you thinking there will be times when the choices are of different types? Cause if not, you could make this a generic class and then you could using objects and take/return the correct types directly.

Reply:
All options (whether choice or numeric) return `IParameter` only and the type conversion is handled by `IParameter.AsType<>` so I suppose we don't need to cast type in option?

Reply:
Makes sense to me.

Reply:
### Resolved